postgresql/resource_postgresql_event_trigger.go

```diff
@@ -0,0 +1,408 @@
+				ValidateFunc: validation.StringInSlice([]string{
+					"disable",
+					"enable",
+					"enable_replica",
+					"enable_always",
+				}, false),
+			},
+			eventTriggerOwnerAttr: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The user name of the owner of the event trigger. You can't use 'current_role', 'current_user' or 'session_user' in order to avoid drifts",
+				ValidateFunc: validation.StringNotInSlice([]string{
+					"current_role",
+					"current_user",
+					"session_user",
+				}, true),
+			},
+		},
+	}
+}
+
+func resourcePostgreSQLEventTriggerCreate(db *DBConnection, d *schema.ResourceData) error {
+	eventTriggerName := d.Get(eventTriggerNameAttr).(string)
+	d.SetId(eventTriggerName)
+
+	b := bytes.NewBufferString("CREATE EVENT TRIGGER ")
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName))
+
+	eventTriggerOn := d.Get(eventTriggerOnAttr).(string)
+	fmt.Fprint(b, " ON ", eventTriggerOn)
+
+	if filters, ok := d.GetOk(eventTriggerFilterAttr); ok {
+		filters := filters.([]interface{})
+
+		for i, filter := range filters {
+			filter := filter.(map[string]interface{})
+
+			if variable, ok := filter[eventTriggerFilterVariableAttr]; ok {
+				if i == 0 {
+					fmt.Fprint(b, " WHEN ", variable)
+				} else {
+					fmt.Fprint(b, " AND ", variable)
+				}
+			}
+
+			if values, ok := filter[eventTriggerFilterValueAttr]; ok {
+				var new_values []string
+
+				for _, value := range values.([]interface{}) {
+					new_values = append(new_values, pq.QuoteLiteral(value.(string)))
+				}
+
+				fmt.Fprint(b, " IN (", strings.Join(new_values, ","), ")")
+			}
+		}
+	}
+
+	eventTriggerFunction := d.Get(eventTriggerFunctionAttr).(string)
+	eventTriggerSchema := d.Get(eventTriggerFunctionSchemaAttr).(string)
+	fmt.Fprint(b, " EXECUTE FUNCTION ", pq.QuoteIdentifier(eventTriggerSchema), ".", eventTriggerFunction, "()")
+
+	createSql := b.String()
+
+	// Enable or disable the event trigger
+	b = bytes.NewBufferString("ALTER EVENT TRIGGER ")
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName))
+
+	eventTriggerEnabled := d.Get(eventTriggerStatusAttr).(string)
+	fmt.Fprint(b, " ", eventTriggerEnabled)
+
+	statusSql := b.String()
+
+	// Table owner
+	b = bytes.NewBufferString("ALTER EVENT TRIGGER ")
+	eventTriggerOwner := d.Get(eventTriggerOwnerAttr).(string)
+	fmt.Fprint(b, pq.QuoteIdentifier(eventTriggerName), " OWNER TO ", eventTriggerOwner)
+
+	ownerSql := b.String()
+
+	// Start transaction
+	txn, err := startTransaction(db.client, d.Get(eventTriggerDatabaseAttr).(string))
+	if err != nil {
+		return err
+	}
+	defer deferredRollback(txn)
+
+	if _, err := txn.Exec(createSql); err != nil {
+		return err
+	}
+
+	if _, err := txn.Exec(statusSql); err != nil {
+		return err
+	}
+
+	if _, err := txn.Exec(ownerSql); err != nil {
+		return err
+	}
+
+	if err := txn.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourcePostgreSQLEventTriggerUpdate(db *DBConnection, d *schema.ResourceData) error {
```

Comment:
You don't manage the renaming of an event properly, similar to the comment on `status` you can test easily by updating your test case with a new step that rename it, e.g.:
```go
    {
        Config: fmt.Sprintf(testAccPostgreSQLEventTriggerConfig, dbName, schemas[0], "test_event"),
        Check: resource.ComposeTestCheckFunc(
            testAccCheckPostgresqlEventTriggerExists("postgresql_event_trigger.event_trigger", dbName),
        ),
    },
    {
        Config: fmt.Sprintf(testAccPostgreSQLEventTriggerConfig, dbName, schemas[0], "test_event_renamed"),
        Check: resource.ComposeTestCheckFunc(
            testAccCheckPostgresqlEventTriggerExists("postgresql_event_trigger.event_trigger", dbName),
        ),
    },
```
In this case, this test will fail with:
```
Error: pq: event trigger "test_event_renamed" does not exist
```


You can simply check here if the name changed with something like:
```go
	if d.HasChange(eventTriggerNameAttr) {
		old, new := d.GetChange(eventTriggerNameAttr)
		if _, err := txn.Exec(
			fmt.Sprintf("ALTER EVENT TRIGGER %s RENAME TO %s", pq.QuoteIdentifier(old), pq.QuoteIdentifier(new)),
		); err != nil {
			return err
		}
	}
```

Reply:
I've just added a rename, I've also added your step to the tests too.